### PR TITLE
Address remaining deprecations

### LINKED ISSUE #308: Outdated code
When I installed it in Atom.io, I got the following alerts from Depreciation Cop:

```
atom.workspaceView is no longer available. In most cases you will not need the view. See the Workspace docs for alternatives: https://atom.io/docs/api/latest/Workspace. If you do need the view, please use atom.views.getView(atom.workspace), which returns an HTMLElement.

Called 1 timeCreate Issue on script repo

Atom.Object.defineProperty.get - C:\Users\Vector\AppData\Local\atom\app-0.182.0\resources\app\src\atom.js:54:11
ScriptOptionsView.initialize - C:\Users\Vector\.atom\packages\script\lib\script-options-view.coffee:51:9
-----
atom.workspaceView is no longer available. In most cases you will not need the view. See the Workspace docs for alternatives: https://atom.io/docs/api/latest/Workspace. If you do need the view, please use atom.views.getView(atom.workspace), which returns an HTMLElement.

Called 1 timeCreate Issue on script repo

Atom.Object.defineProperty.get - C:\Users\Vector\AppData\Local\atom\app-0.182.0\resources\app\src\atom.js:54:11
Object.activate - C:\Users\Vector\.atom\packages\script\lib\script.coffee:30:9
```

### LINKED ISSUE #332: Deprecated code
Got this once I installed. 

`atom.workspaceView is no longer available. In most cases you will not need the view. See the Workspace docs for alternatives: https://atom.io/docs/api/latest/Workspace. If you do need the view, please use atom.views.getView(atom.workspace), which returns an HTMLElement.

Called 1 timeCreate Issue on script repo

Atom.Object.defineProperty.get - C:\Users\User\AppData\Local\atom\app-0.187.0\resources\app\src\atom.js:54:11

ScriptOptionsView.initialize - C:\Users\User\.atom\packages\script\lib\script-options-view.coffee:51:9
.........
atom.workspaceView is no longer available. In most cases you will not need the view. See the Workspace docs for alternatives: https://atom.io/docs/api/latest/Workspace. If you do need the view, please use atom.views.getView(atom.workspace), which returns an HTMLElement.

Called 1 timeCreate Issue on script repo

Atom.Object.defineProperty.get - C:\Users\User\AppData\Local\atom\app-0.187.0\resources\app\src\atom.js:54:11

Object.activate - C:\Users\User\.atom\packages\script\lib\script.coffee:30:9`

### PR DESCRIPTION
**Pull request by erran**
Fixes #308.
Fixes #332.

### CODE REVIEW

**Comment by erran** [comment]
Hey @rgbkrk, I had some extra free time this weekend. I haven't been helping out in maintenance mostly due to the decreased use of Atom I've had (**vi** forever).



Anyways this should close *several* duplicate issues that have been getting filed against atom-script in a while. We should test all use the basic use cases (each type of run, options view, etc.).

**Comment by rgbkrk** [comment]
Niiiiiiiiiiiiiiiiiceeeeee

**Comment by rgbkrk** [comment]
Not sure if it's a regression introduced here, but for known languages that don't have an executable to run we bomb out without our sweet messages:

screenshot 2015-05-03 11 02 04

Additionally, run with options leaves the dialog box up and doesn't run anything when you click "Run". No errors in the console though.

**Comment by erran** [comment]
@rgbkrk It looks like that's in the current release as per https://github.com/rgbkrk/atom-script/issues/275. Let me try to reproduce it locally.

**Comment by erran** [comment]
> Additionally, run with options leaves the dialog box up and doesn't run anything when you click "Run". No errors in the console though.

I'll try to fix this. Definitely new.

**Comment by erran** [comment]
> Additionally, run with options leaves the dialog box up and doesn't run anything when you click "Run". No errors in the console though.
> I'll try to fix this. Definitely new.

@rgbkrk Fixed. Realized that the close button isn't working in the script view though.

**Comment by erran** [comment]
@rgbkrk Scratch that. There are still some issues with it.

**Comment by erran** [comment]
@rgbkrk Should be ready for another set of eyes again. Turns out the dispatch API was a little different than it seemed to be. You add commands to the registry through the selector and dispatch them through a given view.

File: lib/script-view.coffee [hunk 1]
```diff
@@ -267,8 +273,9 @@ class ScriptView extends View
         @pre "PATH: #{_.escape process.env.PATH}"
 
   getCwd: ->
-    if not @runOptions.workingDirectory? or @runOptions.workingDirectory is ''
-      atom.project.getPath()
+    paths = atom.project.getPaths()
+    if paths.length > 0 or not @runOptions.workingDirectory? or @runOptions.workingDirectory is ''
```
Comment (erran) [inline lib/script-view.coffee]: This breaks `getCwd`. :person_frowning:

**Comment by rgbkrk** [comment]
Seems to be working quite well now!

File: lib/script-view.coffee (hunk 1 above)
Comment (rgbkrk) [inline lib/script-view.coffee]: Looks like you addressed it in d7ef205

**Comment by rgbkrk** [comment]
Must say that through all your issue pruning, you are incredible GitHub monster. :smile:



Not sure I know how you did it. Thank you. I'm now going back through to see what I missed. I did make a release but I don't have the fancy releases layout that you had going.

**Comment by erran** [comment]
@rgbkrk Thanks! I ended up using the GitHub search API to get a list of all the issues with the standard deprecation warning stuff in the name. Once I had that I added the list of issue numbers into the commit message to autoclose them. Pretty sure we got them all at this point. Turns out GitHub's issues page has a lot of nice search, labeling, and open/close features too.

I'm trying to get back into open source development, I'm confident I can easily sway my new team into doing it at work. Unfortunately since I don't use Atom as my primary editor anymore I'll probably be contributing a little less here. Maintenance is not a big deal though. It'd be good to update the contributing guidelines on this repository to make contributing easier. I started adding an `easy` label for new contributors to pick up issues with. They're mostly new grammar requests. As for the releases layout, we should look back over the work you did for automating the change log stuff (which is basically what I used for the release text).

**Pull request merged**